src/App.tsx

```diff
@@ -1,24 +1,157 @@
+        .map(tod => setTodos(prev => [tod, ...prev])))
+      .catch(() => setIsError('Unable to load a todo'));
+  }, []);
+
+  const addTodo = ({ userId, title, completed }: Omit<Todo, 'id'>) => {
+    setIsLoading(true);
+
+    postService
+      .createTodos(USER_ID, { userId, title, completed })
+      .then((newTodo: Todo) => {
+        setTodos((prevTodo) => [newTodo, ...prevTodo]);
+        setSelectedId(newTodo.id);
+      })
+      .catch(() => showError('Unable to add a todo'))
+      .finally(() => {
+        setIsLoading(false);
+        setToggleLoader(false);
+      });
+  };
+
+  const deleteTodo = (todoId: number) => {
+    setIsLoading(true);
+    setSelectedId(todoId);
+
+    postService
+      .deleteTodos(todoId)
+      .then(() => setTodos((currentTodos) => currentTodos
+        .filter((todo) => todo.id !== todoId)))
+      .catch(() => showError('Unable to deletion a todo'))
+      .finally(() => {
+        setIsLoading(false);
+        setToggleLoader(false);
+      });
+  };
+
+  const updateTodos = (todoId: number, data: Partial<Todo>) => {
+    setIsLoading(true);
+    setSelectedId(todoId);
+
+    postService
+      .updateTodos(todoId, data)
+      .then((updateTodo) => {
+        setTodos((prev) => prev
+          .map((todo) => (todo.id === todoId ? updateTodo : todo)));
+      })
+      .catch(() => showError('Unable to update a todo'))
+      .finally(() => {
+        setIsLoading(false);
+        setToggleLoader(false);
+      });
+  };
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (todosFilterBy) {
+        case Status.ACTIVE:
+          return !todo.completed;
+
+        case Status.COMPLETED:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+  }, [todosFilterBy, todos]);
+
+  const isActiveTodos = useMemo(() => {
+    return todos.filter((todo) => !todo.completed);
+  }, [todos]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const handlerToggler = () => {
+    setToggleAll(!toggleAll);
+    setToggleLoader(true);
+
+    return filteredTodos
+      .forEach((todo) => updateTodos(todo.id, { completed: toggleAll }));
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          showError={showError}
+          todos={todos}
+          isActive={isActiveTodos.length}
+          isLoading={setIsLoading}
+          createTodo={addTodo}
+          handlerToggler={handlerToggler}
+        />
+
+        {todos.length > 0 && (
```

Comment:
pls [see falsy values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)
```suggestion
        {!!todos.length && (
```